bss/config.go

```diff
@@ -0,0 +1,85 @@
+package bss
+
+import (
+	"time"
+
+	"github.com/urfave/cli"
+
+	"github.com/ethereum-optimism/optimistic-specs/bss/flags"
+)
+
+type Config struct {
+	/* Required Params */
+
+	// L1EthRpc is the HTTP provider URL for L1.
+	L1EthRpc string
+
+	// L2EthRpc is the HTTP provider URL for L2.
+	L2EthRpc string
+
+	// RollupRpc is the HTTP provider URL for the rollup node.
+	RollupRpc string
+
+	// PollInterval is the delay between querying L2 for more transaction
+	// and creating a new batch.
+	PollInterval time.Duration
+
+	// NumConfirmations is the number of confirmations which we will wait after
+	// appending new batches.
+	NumConfirmations uint64
+
+	// SafeAbortNonceTooLowCount is the number of ErrNonceTooLowObservations
+	// required to give up on a tx at a particular nonce without receiving
+	// confirmation.
+	SafeAbortNonceTooLowCount uint64
+
+	// ResubmissionTimeout is time we will wait before resubmitting a
+	// transaction.
+	ResubmissionTimeout time.Duration
+
+	// Mnemonic is the HD seed used to derive the wallet private keys for both
+	// the sequence and proposer. Must be used in conjunction with
+	// SequencerHDPath and ProposerHDPath.
+	Mnemonic string
```

Comment:
I would really like us to not use mnemonics for this type of thing. They should be strictly for backup purposes, and span multiple accounts. We don't want one leaked key to leak all of the keys (if we even use the same mnemonic for the multiple services). It's fine for now, but should be addressed in a next PR before production.

Reply:
We need a tool that can take a mnemonic and then produce encrypted JSON keystores from derived keys

Reply:
In practice each service uses their own mnemonic and the service that needs 2 keys derives both from the same mnemonic. If two unrelated keys were configured and the service was compromised, the security model would be exactly the same